### PR TITLE
[owners] Use a JSON schema validator for OWNERS files

### LINKED ISSUE #912: [owners] Use a proper JSON schema validator
Addresses concerns raised in https://github.com/ampproject/amp-github-apps/pull/911#pullrequestreview-436132725

### PR DESCRIPTION
**Pull request by rcebulko**
Closes #912 right on time, give or take five months :sweat_smile:

### CODE REVIEW

File: owners/test/ownership/parser.test.js [hunk 1]
```diff
@@ -556,33 +569,33 @@ describe('owners parser', () => {
...
-        it('does not produce a reviewer set rule', () => {
+        it('does not produce a result', () => {
           const {result} = parser.parseOwnersFileDefinition('OWNERS', fileDef);
-          expect(result.length).toEqual(0);
+          expect(result).toEqual([]);
         });
       });
 
-      describe('for an unknown team name', () => {
+      describe('for an unknown team namyname', () => {
```
Comment (danielrozenberg) [inline owners/test/ownership/parser.test.js:595]: namyname?

File: owners/test/ownership/parser.test.js [hunk 2]
```diff
@@ -594,7 +607,7 @@ describe('owners parser', () => {
 
         it('does not produce a reviewer set rule', () => {
           const {result} = parser.parseOwnersFileDefinition('OWNERS', fileDef);
-          expect(result.length).toEqual(0);
+          expect(result.length).toEqual(1);
```
Comment (danielrozenberg) [inline owners/test/ownership/parser.test.js:610]: This seems like an odd change

File: owners/test/ownership/parser.test.js [hunk 3]
```diff
@@ -245,105 +259,102 @@ describe('owners parser', () => {
...
     describe('when an unexpected key is present', () => {
       it('reports an error', () => {
-        const {errors} = parser._parseRuleDefinition('', {
+        const {errors} = parseRuleDefinition({
           owners: [{name: 'blah'}],
           notify: true, // This should be in the owner, not the rule
         });
         expect(errors[0].message).toEqual(
-          'Unexpected key(s) "notify" in rule definition'
+          '`OWNERS.rules[0]` should NOT have additional properties'
```
Comment (danielrozenberg) [inline owners/test/ownership/parser.test.js:279]: nit: is it possible to set custom error messages in the schema so that the context stays?

**Review by danielrozenberg — APPROVED** [review]

File: owners/test/ownership/parser.test.js (hunk 3 above)
Comment (rcebulko) [reply owners/test/ownership/parser.test.js:279]: No :disappointed: The best I could do is post-process the error message, which would add almost as much logic as this PR removes. I definitely dislike that aspect; a lot of clarity is lost when something is invalid

File: owners/test/ownership/parser.test.js (hunk 2 above)
Comment (rcebulko) [reply owners/test/ownership/parser.test.js:610]: The test data used to have an empty list of rules (out of laziness) which was allowed, but now the validator yells if the set of rules or owners doesn't have at least one entry, so the test data now has 1 rule.

File: owners/test/ownership/parser.test.js (hunk 1 above)
Comment (rcebulko) [reply owners/test/ownership/parser.test.js:595]: Whoops lol

**Pull request merged**